### DISCUSSION TITLE
How to modify the build.gradle file in nativescript?

### DISCUSSION THREAD

**Discussion by tbontb-iaq** [NativeScript/NativeScript#9955] [Q&A]
I just want to try to run a new project I created via `ns create` but I get the following error:
````
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':extractReleaseAnnotations'.
> Could not resolve all files for configuration ':detachedConfiguration2'.
   > Could not resolve com.android.tools.lint:lint-gradle:30.1.2.
     Required by:
         project :
      > Could not resolve com.android.tools.lint:lint-gradle:30.1.2.
         > Could not get resource 'https://maven.google.com/com/android/tools/lint/lint-gradle/30.1.2/lint-gradle-30.1.2.pom'.
            > Could not GET 'https://maven.google.com/com/android/tools/lint/lint-gradle/30.1.2/lint-gradle-30.1.2.pom'.
               > Connect to maven.google.com:443 [maven.google.com/142.251.42.238] failed: connect timed out

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 36s
Failed to build plugin @nativescript/core :
Error: Command ./gradlew failed with exit code 1
````
This is because in some countries or regions, such as China, access to some google sites is prohibited by the operator.
The proxy I set for nativescript is not working in gradle, I think I should set it in gradle's config file.
From what I have searched, the solution is to change `https://maven.google.com` in `build.gradle` to `https://dl.google.com/dl/android/maven2/`, so how do I do this in nativescript, I noticed a [stackoverflow question](https://stackoverflow.com/questions/38094389/nativescript-how-to-avoid-editing-build-gradle-in-platforms-android), but it doesn't work for me.

**Comment by tbontb-iaq** [comment]
Also, the nativescript version I am using is 8.2.3.
I tried creating the `build.gradle` file under `App_Resources/Android` and modifying the `build.gradle` file in the `platforms/android` directory, but the generated files in the `tempPlugin/core` directory do not have what I expected the result of.
I used the same method to create and modify the `gradle.properties` file to set the proxy, same failed.
Even more annoying is that the global proxy I set in `<user>/.gradle/gradle.properties` doesn't work either.

**Comment by rigor789** [comment]
I believe you can make those changes in `App_Resources/Android/before-plugins.gradle`. The plugins are currently built using a config managed in the `nativescript-cli`, but we do want to/plan to move to using the same setup coming from the runtime (`@nativescript/android`).

**Comment by DuckBurnIncense** [comment]
鼓捣了半天瞎猫碰到死耗子终于鼓捣出来了 (鼓捣出的结果如下图, 那个绿色的 `buildscript.gradle` 是我翻旧版文档搞的, 好像并没有什么用).

![图像 002](https://user-images.githubusercontent.com/87637270/210264220-cdc95625-79de-440e-bca7-f3c8e881c738.png)

~~emmm果断去学原生开发了~~
